### PR TITLE
Account interface that works with overlays.

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

Account interface handles last_id and signature dup verification. It should only handle accounts, and it needs to work with overlays for checkpointing and forking.

#### Summary of Changes

* Move last_id/sig dup checks out of accounts
* handle overlays in the load apis

Fixes #

### CODE REVIEW

File: src/bank.rs [hunk 1]
```diff
@@ -159,13 +159,14 @@ impl Bank {
             mint_account.tokens -= genesis_block.bootstrap_leader_tokens;
             bootstrap_leader_account.tokens += genesis_block.bootstrap_leader_tokens;
             self.accounts.store_slow(
+                true,
```
Comment (garious) [inline src/bank.rs]: It's unusual to see a hardcoded boolean passed to every instance of a function. Usually that's an indicator that there should be 2 (sometimes 3) separate functions. If the branch you'll always take is statically known, why pass a runtime parameter?  Also, no test for that new branch?

Comment (aeyakovenko) [reply src/bank.rs]: @garious added test.  The bank will use this parameter once its checkpointing its state, so right now Accounts implements an expanded interface which is a superset of what the old bank needs and what is necessary for checkpointing.

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

File: src/accounts.rs [hunk 2]
```diff
@@ -81,23 +78,36 @@ impl AccountsDB {
...
         None
     }
-
-    pub fn store(&mut self, pubkey: &Pubkey, account: &Account) {
+    /// Store the account update.  If the update is to delete the account because the token balance
+    /// is 0, purge needs to be set to true for the delete to occur in place.
+    pub fn store(&mut self, purge: bool, pubkey: &Pubkey, account: &Account) {
         if account.tokens == 0 {
-            self.accounts.remove(pubkey);
+            if purge {
```
Comment (rob-solana) [inline src/accounts.rs]: purge is hurting my brain...  is that only set on collapse of a fork?

File: src/bank.rs [hunk 3]
```diff
@@ -178,7 +179,7 @@ impl Bank {
             loader: solana_native_loader::id(),
         };
         self.accounts
-            .store_slow(&system_program::id(), &system_program_account);
+            .store_slow(true, &system_program::id(), &system_program_account);
```
Comment (rob-solana) [inline src/bank.rs]: if you store(tokens=0) with purge=true onto a fork with a parent, it exposes the parent's state, right?

File: src/accounts.rs (hunk 2 above)
Comment (aeyakovenko) [reply src/accounts.rs]: @rob-solana, yea when merging into root it’s set to true. I could probably hide it better.

Comment (rob-solana) [reply src/accounts.rs]: you could accept the same &[U] argument as load() and deduce purge by checkpoints.len()?

**Comment by aeyakovenko** [comment]
part of #2289

**Pull request merged**